libsolidity/parsing/Parser.cpp

```diff
@@ -94,14 +94,22 @@ ASTPointer<SourceUnit> Parser::parse(CharStream& _charStream)
 		m_recursionDepth = 0;
 		m_scanner = make_shared<Scanner>(_charStream);
 		ASTNodeFactory nodeFactory(*this);
+		bool finishedParsingPragmas = false;
+		m_experimentalSolidityEnabledInCurrentSourceUnit = false;
 
 		vector<ASTPointer<ASTNode>> nodes;
 		while (m_scanner->currentToken() != Token::EOS)
 		{
+			// Encountering the first non pragma token indicates that we've finished parsing pragmas
+			// at the beginning of the source unit, and can therefore no longer accept subsequent "experimental
+			// solidity" pragmas (all other pragmas are will be accepted though).
+			if (m_scanner->currentToken() != Token::Pragma)
+				finishedParsingPragmas = true;
```

Comment:
Uh, the best name would probably be `finishedParsingTopLevelPragmas` because that's exactly what the flag indicates. `unexpectedExperimentalPragma` doesn't really make sense, because we set the value to `true` as soon as we detect a non pragma token (e.g. `contract`, `struct`, free function definition, etc.), so that name doesn't really fit.

Reply:
I won't hold up the PR because of it, but I'd still argue that
```
while (m_scanner->currentToken() == Token::Pragma)
    nodes.push_back(parsePragmaDirective(false));
while (m_scanner->currentToken() != Token::EOS)
{
...
    nodes.push_back(parsePragmaDirective(true));
...
}
```
is 10 times simpler for this purpose and would be able to live without any comments :-).

Reply:
> I won't hold up the PR because of it, but I'd still argue that
> 
> ```
> while (m_scanner->currentToken() == Token::Pragma)
>     nodes.push_back(parsePragmaDirective(false));
> while (m_scanner->currentToken() != Token::EOS)
> {
> ...
>     nodes.push_back(parsePragmaDirective(true));
> ...
> }
> ```
> 
> is 10 times simpler for this purpose and would be able to live without any comments :-).

FINE